File: src/main/java/net/hectus/neobb/turn/default_game/throwable/TSplashLevitationPotion.java
package net.hectus.neobb.turn.default_game.throwable;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import net.hectus.neobb.util.ItemBuilder;
import org.bukkit.Color;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.entity.Projectile;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.PotionMeta;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TSplashLevitationPotion extends ThrowableTurn implements BuffFunction, SupernaturalClazz {
    public TSplashLevitationPotion(NeoPlayer player) { super(player); }
    public TSplashLevitationPotion(Projectile data, Location impact, NeoPlayer player) { super(data, impact, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public ItemStack item() {
        return new ItemBuilder(Material.SPLASH_POTION)
                .editMeta(meta -> ((PotionMeta) meta).addCustomEffect(new PotionEffect(PotionEffectType.LEVITATION, 200, 0, true), true))
                .editMeta(meta -> ((PotionMeta) meta).setColor(Color.WHITE))
                .build();
    }

    @Override
    public void apply() {
        if (player.player.hasPotionEffect(PotionEffectType.LEVITATION))
            player.removeModifier("extra-turn");

        player.game.turnScheduler.runTaskLater("levitation", () -> player.opponents(true).forEach(p -> {
            if (p.player.hasPotionEffect(PotionEffectType.LEVITATION))
                player.game.eliminatePlayer(p);
        }), 3);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn());
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/throwable/TSplashLevitationPotion.kt
package net.hectus.neobb.turn.default_game.throwable

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.ItemBuilder
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.util.Modifiers
import org.bukkit.Color
import org.bukkit.Material
import org.bukkit.entity.Projectile
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.ItemMeta
import org.bukkit.inventory.meta.PotionMeta
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

class TSplashLevitationPotion(data: Projectile?, cord: Cord?, player: NeoPlayer?) : ThrowableTurn(data, cord, player), BuffFunction, SupernaturalClazz {
    override val cost: Int = 4

    override fun item(): ItemStack = ItemBuilder(Material.SPLASH_POTION)
        .editMeta { meta: ItemMeta -> (meta as PotionMeta).addCustomEffect(PotionEffect(PotionEffectType.LEVITATION, 200, 0, true), true) }
        .editMeta { meta: ItemMeta -> (meta as PotionMeta).color = Color.WHITE }
        .build()

    override fun apply() {
        if (player!!.player.hasPotionEffect(PotionEffectType.LEVITATION)) player.removeModifier(Modifiers.Player.EXTRA_TURN)

        player.game.turnScheduler.runTaskLater(ScheduleID.LEVITATION, 3) {
            player.opponents(true).forEach { p ->
                if (p.player.hasPotionEffect(PotionEffectType.LEVITATION))
                    player.game.eliminate(p)
            }
        }
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}